crates/core/tests/client.rs

```diff
@@ -27,4 +30,108 @@ mod tests {
             ))
         )
     }
+
+    #[tokio::test]
+    async fn test_starknet_event_to_eth_log() {
+        let client = setup_mock_client_crate().await;
+        // given
+        let data =
+            vec![0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10];
+        let felt_data: Vec<FieldElement> =
+            data.iter().map(|&x| FieldElement::from_dec_str(&x.to_string()).unwrap()).collect();
+        let bytes_data: Bytes = felt_data.iter().flat_map(|felt| felt.to_bytes_be()).collect::<Vec<u8>>().into();
+        // see https://github.com/kkrt-labs/kakarot/blob/2133aaf58d5c8ae493c579570e43c9e011774309/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py#L120 this test generates the starknet event and ethereum log expected pair
+        let event3 = Event {
+            from_address: FieldElement::from_hex_be("0x2B61c43A85bD35987C5311215e8288b823A6873E").unwrap(),
```

Comment:
this address is not the one that shows up in the raw_event, but the self.address that shows up here https://github.com/kkrt-labs/kakarot/blob/2133aaf58d5c8ae493c579570e43c9e011774309/tests/utils/contracts.py#L123

this value is set during a deploy flow, assumedly something like this https://github.com/kkrt-labs/kakarot/blob/2133aaf58d5c8ae493c579570e43c9e011774309/scripts/utils/kakarot.py#L105

so are technically skipping testing the `get_evm_address` of the starknet address component of this process, for now, because of the first comment issue with calling `get_evm_address` or `safe_get_evm_address`

i also had the thought:

- we changed how events are emitted to ensure they only happen after an execution context is successfully stopped: https://github.com/kkrt-labs/kakarot/blob/2133aaf58d5c8ae493c579570e43c9e011774309/src/kakarot/execution_context.cairo#L317
- would that change effect the `from_address` in the starknet event?
- if so, a solution would be to add to the eth_contract interface an 'emit events' method that the vm calls into at this point, instead of emitting the events themselves.

Reply:
elias' poetry about the potential issue:

```
u could make it so that

the first key

is always

the evm address

of the execution context “true caller”

of the event
```